crates/cargo-contract/src/cmd/extrinsics/upload.rs

```diff
@@ -145,12 +149,11 @@ impl UploadCommand {
 
         let result = submit_extrinsic(client, &call, signer).await?;
 
-        display_events(
-            &result,
-            transcoder,
-            &client.metadata(),
-            &self.extrinsic_opts.verbosity()?,
-        )?;
+        let call_result =
```

Comment:
In case of an error returned from `submit_extrinsic`, we should also display the error as json, just like we do in the `call` extrinsic.